packages/modelserver-node/src/client/model-server-client.ts

```diff
@@ -130,6 +130,7 @@ export interface ExecuteCommand {
 export interface ExecutePatch {
     type: 'modelserver.patch';
     data: Operation | Operation[];
+    format?: Format | string;
```

Comment:
This is the first instance that we have of the `format` being a property of the message and not a request URI query parameter. Consequently, the Java server will receive this `format` property. Is it being updated to do something with it? Because the Java server will have to know how to decode objects in these messages, and I think currently it assumes the `json-v2` format.

I would have expected rather that the `GET` request to open a transaction would add a `format` parameter to govern all messages sent to and from the Java server, so that then it could do proper decoding of incoming messages and encoding of its outgoing reply messages.

What do you think? In either case, I think there will need to be changes in the Java server, or did I miss that all of the format translation is being done in the Node layer here and so the Java server will only ever see `json-v2` format?

Reply:
On further reflection, there should not need to be any impact here on the Java server. If it isn't already, then the Node layer as suggested by my previous comment can translate anything that it forwards to the Java layer into `json-v2` format. So that introducing the format support into the Java layer can be left for the eventual case of a client that doesn't use the Node layer needing the same format capability.

Hopefully the Node layer would translate incoming transaction messages into `json-v2` format for internal processing, anyways? It needs some singular consistent internal representation ...

Reply:
Thanks for your further feedback!
I updated (and already squashed) my branch and for now, the modelserver-node layer will interpret the received patches as `ModelServerMessage`s (directly from `json-v2`) and possibly encode with the custom format beforehand if requested. So for now, it should be sufficient to offer this possibility on the modelserver-node layer.
Would be great if you could have another look, thanks!